Dapper/SqlMapper.GridReader.Async.cs

```diff
@@ -94,6 +121,42 @@ private Task<IEnumerable<T>> ReadAsyncImpl<T>(Type type, bool buffered)
                 }
             }
 
+            private Task<T> ReadFirstOrDefaultAsyncImpl<T>(Type type)
+            {
+                var dbReader = reader as DbDataReader;
+                if (dbReader != null) return ReadFirstOrDefaultAsyncImplImpl<T>(dbReader, type);
+
+                // no async API available; use non-async and fake it
+                return Task.FromResult<T>(ReadFirstOrDefaultImpl<T>(type));

```

Comment:
The reason this *specific* method exists is so that we don't have a redundant `return await SomethingThatReturnsATask();`, because that unnecessary wrapping annoys me